linkerd/app/test/src/http_util.rs

```diff
@@ -93,11 +93,15 @@ where
     let (client_io, proxy) = run_proxy(server).await;
     let (client, client_bg) = connect_client(client_settings, client_io).await;
     let bg = async move {
-        let res = tokio::try_join! {
+        let proxy = async move { proxy.await.expect("proxy background task panicked") };
+        let client_bg = async move { client_bg.await.expect("client background task panicked") };
+        let res = tokio::join! {
             proxy,
             client_bg,
         };
-        res.unwrap();
+        res.0?;
+        res.1?;
+        Ok(())
```

Comment:
what's going on here? given that these are join handles, why do we need to stash these in async closures and use join? I think this can be:

```suggestion
         proxy
             .await
             .expect("proxy task panicked")
             .await?;
         client_bg
             .await
             .expect("client task panicked")
             .await?;
         Ok(())
```

Reply:
IIRC, I used `join` ages ago because I wasn't sure which task would complete first, and I didn't want it to hang...awaiting them in sequence is probably fine. They're just `JoinHandle<Result<...>>`, though, not `JoinHandle<impl Future<...>>`, so the second `await` in your example isn't necessary.

Reply:
...i guess they're spawned so it doesn't actually matter.

Reply:
Updated the suggestion to have a single await

Reply:
yeah, i checked and it turns out i was totally overthinking it whenever it was that i wrote this originally, the join is completely unnecessary.